libbeat/common/match/matchers.go

```diff
@@ -251,14 +251,19 @@ func (m *matchAny) Match(_ []byte) bool       { return true }
 func (m *matchAny) MatchString(_ string) bool { return true }
 func (m *matchAny) String() string            { return "<any>" }
 
-func bytesToString(b []byte) string {
-	bh := (*reflect.SliceHeader)(unsafe.Pointer(&b))
-	sh := reflect.StringHeader{Data: bh.Data, Len: bh.Len}
-	return *(*string)(unsafe.Pointer(&sh))
-}
-
-func stringToBytes(s string) []byte {
-	sh := (*reflect.StringHeader)(unsafe.Pointer(&s))
-	bh := reflect.SliceHeader{Data: sh.Data, Len: sh.Len, Cap: sh.Len}
-	return *(*[]byte)(unsafe.Pointer(&bh))
+func bytesToString(b []byte) (s string) {
+	pb := (*reflect.SliceHeader)(unsafe.Pointer(&b))
+	ps := (*reflect.StringHeader)(unsafe.Pointer(&s))
+	ps.Data = pb.Data
+	ps.Len = pb.Len
+	return
+}
```

Comment:
Alternative oneline: `*(*string)(unsafe.Pointer(&b))`

e.g. from stdlib https://github.com/golang/go/blob/master/src/strings/builder.go#L48

Reply:
```
BenchmarkStringToBytes
BenchmarkStringToBytes-16            	1000000000	         0.2325 ns/op	       0 B/op	       0 allocs/op
BenchmarkStringToBytesOneLiner
BenchmarkStringToBytesOneLiner-16    	1000000000	         0.2338 ns/op	       0 B/op	       0 allocs/op
```
seems the same, but sure

Reply:
```
BenchmarkBytesToString
BenchmarkBytesToString-16            	1000000000	         0.2419 ns/op	       0 B/op	       0 allocs/op
BenchmarkBytesToStringOneLiner
BenchmarkBytesToStringOneLiner-16    	1000000000	         0.2455 ns/op	       0 B/op	       0 allocs/op
```